src/main/java/com/hillel/items_exchange/service/impl/LocationServiceImpl.java

```diff
@@ -74,4 +91,59 @@ public LocationDto update(LocationDto locationDto) {
         return convertTo(updatedLocation, LocationDto.class);
     }
-}
\ No newline at end of file
+
+    @Override
+    public String createFileToInitLocations(String creatingData)
+            throws IOException, InvalidLocationInitFileCreatingDataException {
+        File file = new File(PATH_LOCATION_INIT_FILE);
+        List<Location> locations = mapCreatingDataToLocations(creatingData);
+        String initString = "INSERT INTO `evo_exchange`.`location` (`id`, `city`, `district`, `area`, `i18n`) VALUES";
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(file, false))) {
+            writer.append(initString);
+            for (int i = 0; i < locations.size(); i++) {
+                Location location = locations.get(i);
+                writer.append(String.format(" ('%s', '%s','%s','%s','%s')",
+                        i + 1,
+                        location.getCity().replace("'", "\\'"),
+                        location.getDistrict().replace("'", "\\'"),
+                        location.getArea().replace("'", "\\'"),
+                        location.getI18n()));
+                if (i < locations.size()-1) writer.append(",");
+            }
+        }
+        return Files.readString(Path.of(file.getAbsolutePath()));
+    }
+
+    @Override
+    public boolean isLocationDataValid(String locationDataString) {
+        Pattern pattern = Pattern.compile(LOCATION_INIT_FILE_CREATE_DATA_PATTERN);
+        return Arrays.stream(locationDataString.split("},"))
+                .map(s -> s.replace("[", ""))
+                .map(s -> s.replace("]", ""))
+                .map(s -> s.replace("{", ""))
+                .map(s -> s.replace("}", ""))
+                .allMatch(s -> pattern.matcher(s).matches());
+    }
+
+    private List<Location> mapCreatingDataToLocations(String creatingData)
+            throws InvalidLocationInitFileCreatingDataException, JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
+        if (!isLocationDataValid(creatingData)) throw new InvalidLocationInitFileCreatingDataException(
+                getExceptionMessageSource("exception.invalid.locations.file.creating.data"));
+        List<String> stringsToMap = Arrays.stream(creatingData.split("},"))
+                .map(s -> s.replace("[", ""))
+                .map(s -> s.replace("]", ""))
+                .map(s -> s.replace("{\"en\":", "{\"lang\":\"EN\""))
+                .map(s -> s.replace("\"ua\":", "{\"lang\":\"UA\""))
+                .map(s -> s.replace("\"ru\":", "{\"lang\":\"RU\""))
+                .map(s -> s + "}")
+                .map(s -> s.replace("}}", "}"))
+                .map(s -> s.replace("{\"city\"", ",\"city\""))
+                .collect(Collectors.toList());
+        List<Location> result = new ArrayList<>();
+        for (String s : stringsToMap) {
+            result.add(mapper.readValue(s, Location.class));
+        }
+        return result;
```

Comment:
You could refactor lines from 143 up to 147 using stream and with collecting to list.